transferui/src/androidTest/java/com/hyperwallet/android/ui/transfer/TransferUserFundsTest.java

```diff
@@ -124,44 +152,625 @@ public void testTransferFunds_verifyAddDestinationDisplayedWhenUserHasNoExternal
 
         onView(withId(R.id.transfer_summary)).check(matches(not(isDisplayed())));
         onView(withId(R.id.transfer_action_button)).check(matches(isDisplayed()));
-        onView(withId(R.id.transfer_action_button)).check(matches(not(isEnabled())));
+        onView(withId(R.id.transfer_action_button)).check(matches(isEnabled()));
+    }
+
+    @Test
+    public void testTransferFunds_createTransferWithFX() throws InterruptedException {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("transfer_method_list_single_bank_account_cad_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("create_transfer_quote_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("create_transfer_fx_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("schedule_transfer_success_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        final CountDownLatch gate = new CountDownLatch(1);
+        final BroadcastReceiver br = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                gate.countDown();
+
+                StatusTransition transition = intent.getParcelableExtra(
+                        "hyperwallet-local-broadcast-payload");
+                assertThat("Token is incorrect", transition.getToken(), is("sts-2157d925-90c9-407b-a9d6-24a0d9dacfb6"));
+                assertThat("To Status is incorrect", transition.getToStatus(), is("SCHEDULED"));
+                assertThat("From Status is incorrect", transition.getFromStatus(), is("QUOTED"));
+                assertThat("Transition is incorrect", transition.getTransition(), is("SCHEDULED"));
+                assertThat("Created on is incorrect", transition.getCreatedOn(), is("2019-08-12T17:39:35"));
+
+            }
+        };
+
+        LocalBroadcastManager.getInstance(mActivityTestRule.getActivity().getApplicationContext())
+                .registerReceiver(br, new IntentFilter("ACTION_HYPERWALLET_TRANSFER_CREATED"));
```

Comment:
can you work with @peter-joseph to update the action name to ACTION_HYPERWALLET_TRANSFER_SCHEDULED?

Reply:
Do we need one for created and one for scheduled?